src/js/view/utils/resize-listener.js

```diff
@@ -1,44 +1,56 @@
 import { requestAnimationFrame, cancelAnimationFrame } from 'utils/request-animation-frame';
 import { createElement } from 'utils/dom';
-import { css, style } from 'utils/css';
+import { style } from 'utils/css';
 
-const topLeft = {
-    display: 'block',
-    position: 'absolute',
-    top: 0,
-    left: 0
-};
+const instances = [];
+let resizeRaf = -1;
 
-const stretch = {
-    width: '100%',
-    height: '100%'
-};
-
-let contractTriggerCssAdded = false;
-
-const lazyAddCss = function() {
-    if (!contractTriggerCssAdded) {
-        contractTriggerCssAdded = true;
-        css('.jw-contract-trigger::before', Object.assign({
-            content: '',
-            overflow: 'hidden',
-            width: '200%',
-            height: '200%'
-        }, topLeft));
-    }
-};
+function scrollListener() {
+    cancelAnimationFrame(resizeRaf);
+    resizeRaf = requestAnimationFrame(() => {
+        instances.forEach(resizeListener => {
+            resizeListener.updateBounds();
+        });
+        instances.forEach(resizeListener => {
+            resizeListener.contractElement.scrollLeft = resizeListener.width * 2;
```

Comment:
assuming this is faster/less thrashy than having one `instances.forEach` block which does all the scroll/style updates for each listener in a single go. and also assuming that's because only one of these would actually force layout (style) and as a result we want to separate it from the compute of the scroll left? though that would at least paint right?

Reply:
Seems like we pay mostly on the last `scrollLeft` update after a `style` update. Batching prevent thrashing between the back and forth of setting both for each player.

Reply:
@jnatalzia I just pushed some updates that remove the `scheduleResponsiveRedraw` callback and instead implement it in the scroll listener.

This is now a view resize listener, and the player and widget views both implement `updateBounds`, `updateStyles` (handled when updating the scroll listener styles) and `checkResized`.

This is a nice boost for both single and multiple players since it removes the thrashing caused by the read writes of the resize-listener followed by the read writes of `scheduleResponsiveRedraw`.